src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -393,40 +409,95 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
            if( .not. parallelization_over_ensmembers )then
               if (mype == 0) write(6,'(a,a)') &
                  'CALL READ_FV3_REGIONAL_ENSPERTS FOR ENS DATA with the filename str : ',trim(ensfilenam_str)
-              if (.not. (l_use_dbz_directDA .or. if_model_dbz) ) then ! Read additional hydrometers and w for dirZDA
-                 call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz)
-              else
-                 if( l_use_dbz_directDA ) then
-                    call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
-                                                        g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w)
-                 else if( if_model_dbz )then
-                    call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
-                                                        g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_dbz=dbz)
-                 end if
-              end if
+          
+              ! There are three options to control the list of variables that
+              ! will be read in along with the basic variables, ps,u,v,tv,rh,oz.
+              ! Here the 6 cases that are considered in
```

Comment:
Much cleaner organization.  Just maybe a little note could be added :  l_use_dbz_directDA and if_modle_dbz couldn't be true at the same time. hence, all possible valid set up from the 3 options are 6 cases , which are treated as below.